### PR TITLE
Add explanation of loss function and optimization

### LINKED ISSUE #251: Add more explanation on how loss function and optimization works
It would be nice to add a bit more theory on loss functions and optimization. We often get questions about it and explain along the way by improvising, but it would be nice if there is some material about it with pictures. Something like: 
image
and
image

This would fit in https://carpentries-incubator.github.io/deep-learning-intro/02-keras/index.html#5-choose-a-loss-function-and-optimizer

### PR DESCRIPTION
**Pull request by svenvanderburg**
Fixes #251 #106

### CODE REVIEW

**Comment by svenvanderburg** [comment]
Closing in favour of #352 using the new template
